src/Compositor.cpp

```diff
@@ -1128,6 +1130,35 @@ void CCompositor::focusWindow(PHLWINDOW pWindow, SP<CWLSurfaceResource> pSurface
     if (pWindow && pWindow->m_isX11 && pWindow->isX11OverrideRedirect() && !pWindow->m_xwaylandSurface->wantsFocus())
         return;
 
+    const bool CONFLICTSWITHFULLSCREEN = pWindow && pWindow->m_workspace && pWindow->m_workspace->m_hasFullscreenWindow && pWindow->m_workspace->getFullscreenWindow() != pWindow;
```

Comment:
I don't like this move. `focusWindow` should only be an internal function, FS conflicts should be done elsewhere. Perhaps make a new fn for a user-requested focus that does the check?

Reply:
Like this?

I still don't quite understand the logical separation of responsibilities for the functions, so could you help me with their names?